Tests/MessagingPush/PushHandling/PushEventHandlerProxyTest.swift

```diff
@@ -28,7 +28,7 @@ class PushEventHandlerProxyTest: UnitTest {
             delegate1.onPushActionClosure = { _, completion in
                 expectDelegatesReceiveEvent.fulfill()
 
-                self.runOnMain {
+                self.runOnBackground {
```

Comment:
shouldn't notification callback be running on main?

Reply:
The `delegate1`  calling the completion handler represents a 3rd party push handler. They can call the completion handler on whatever thread they want. Our SDK can call `delegate1`  on the main thread, but the delegate gets to decide what thread they complete on.